types/jquery/misc.d.ts

```diff
@@ -4707,114 +4631,1733 @@ $( "ul" ).click( handler ).find( "ul" ).hide();
+        mouseover: MouseOverEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        mouseup: MouseUpEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+
+        // KeyboardEvent
+
+        keydown: KeyDownEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        keypress: KeyPressEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        keyup: KeyUpEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+
+        // TouchEvent
+
+        touchcancel: TouchCancelEvent<TDelegateTarget, TData, TCurrentTarget>;
+        touchend: TouchEndEvent<TDelegateTarget, TData, TCurrentTarget>;
+        touchmove: TouchMoveEvent<TDelegateTarget, TData, TCurrentTarget>;
+        touchstart: TouchStartEvent<TDelegateTarget, TData, TCurrentTarget>;
+
+        // FocusEvent
+
+        blur: BlurEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        focus: FocusEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        focusin: FocusInEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+        focusout: FocusOutEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
+
+        [type: string]: TriggeredEvent<TDelegateTarget, TData, TCurrentTarget, TTarget>;
     }
-}
-```
-         */
-        filter(event: Event, originalEvent: _Event): void;
-    } | {
-        [key: string]: never;
-    };
 
-    /**
-     * The `fixHooks` interface provides a per-event-type way to extend or normalize the event object that jQuery creates when it processes a _native_ browser event.
-     * @see \`{@link https://learn.jquery.com/events/event-extensions/#jquery-event-fixhooks-object }\`
-     */
-    interface FixHooks {
-        [event: string]: FixHook;
+    // Extra parameters can be passed from trigger()
+    type EventHandlerBase<TContext, T> = (this: TContext, t: T, ...args: any[]) => any;
+
+    type EventHandler<
+        TCurrentTarget,
+        TData = undefined
+    > = EventHandlerBase<TCurrentTarget, TriggeredEvent<TCurrentTarget, TData>>;
+
+    type TypeEventHandler<
+        TDelegateTarget,
+        TData,
+        TCurrentTarget,
+        TTarget,
+        TContext,
+        TType extends keyof TypeToTriggeredEventMap<TDelegateTarget, TData, TCurrentTarget, TTarget>
+    > = EventHandlerBase<TContext, TypeToTriggeredEventMap<TDelegateTarget, TData, TCurrentTarget, TTarget>[TType]>;
+
+    interface TypeEventHandlers<
+        TDelegateTarget,
+        TData,
+        TCurrentTarget,
+        TTarget,
+        TContext
+    > extends _TypeEventHandlers<TDelegateTarget, TData, TCurrentTarget, TTarget, TContext> {
+        // No idea why it's necessary to include `object` in the union but otherwise TypeScript complains that
+        // derived types of Event are not assignable to Event.
+        [type: string]: TypeEventHandler<TDelegateTarget, TData, TCurrentTarget, TTarget, TContext, string> |
+                        false |
+                        undefined |
+                        object;
```

Comment:
Adding `object` to the union weakens the type a bit, although in practice, it's unlikely that users will attempt to a assign a non-function object to a property. I don't understand what's causing the assignability issue nor do I understand why including `object` in the union fixes it.